### PR TITLE
HPCC-27762 File urls incorrect on roxie query details page of ECL Watch 9

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/Activities.tsx [hunk 1]
```diff
@@ -110,7 +110,7 @@ export const Activities: React.FunctionComponent<ActivitiesProps> = ({
                 formatter: function (_gid, row) {
                     if (activity.isInstanceOfWorkunit(row)) {
                         if (row.GraphName) {
-                            return `<a href='#/graphs/${row.GraphName}/${row.GID}' class='dgrid-row-url2'>${row.GraphName}-${row.GID}</a>`;
+                            return <Link href={`#/graphs/${row.GraphName}/${row.GID}`}>{row.GraphName}-{row.GID}</Link>;
```
Comment (jeclrsg) [inline esp/src/src-react/components/Activities.tsx:113]: @GordonSmith Aside from changes in QueryLogicalFiles and QuerySuperFiles which were the actual fix required, I also noticed a handful of files that had string interpolation for the anchor tag in formatter functions, so thought I would change those out for <Link>s in this PR as well.

File: esp/src/src-react/components/QuerySuperFiles.tsx [hunk 2]
```diff
@@ -42,7 +42,7 @@ export const QuerySuperFiles: React.FunctionComponent<QuerySuperFilesProps> = ({
             File: {
                 label: nlsHPCC.File,
                 formatter: function (item, row) {
-                    return `<a href="#/files/${querySet}/${item}">${item}</a>`;
+                    return <Link href={`#/files/${item}`}>{item}</Link>;
```
Comment (GordonSmith) [inline esp/src/src-react/components/QuerySuperFiles.tsx:45]: Dropping the querySet is on purpose (which seems sensible)?

**Review by GordonSmith — COMMENTED** [review]
@jeclrsg just one minor question.

File: esp/src/src-react/components/QuerySuperFiles.tsx (hunk 2 above)
Comment (jeclrsg) [reply esp/src/src-react/components/QuerySuperFiles.tsx:45]: Yes, that's correct, with the querySet it wasn't a valid route.

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**